dev/release/download_release.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python
```

Comment:
Is this based on https://github.com/apache/arrow/blob/master/dev/release/download_rc_binaries.py ?
How about improving apache/arrow's one and reuse it from apache/arrow-adbc?

Reply:
https://github.com/apache/arrow/pull/14848